porcupine/default_filetypes.toml

```diff
@@ -30,13 +30,48 @@
 #        Arguments starting with '-' are ignored, so the regex '^#!/bin/foo$'
 #        matches the shebang '#!/bin/foo --bar --baz'.
 #
+#    syntax_highlighter
+#        Porcupine currently comes with two syntax highlighters that this
+#        setting chooses from:
+#
+#            "pygments" (default)
+#                This syntax highlighter supports many different languages and
+#                it is easy to configure, but it is slow with large files. It
+#                also has known bugs that are basically impossible to fix:
+#                sometimes editing the code and reverting your edits changes how
+#                the code is highlighted.
+#
+#                Recommended for any language with small files.
+#
+#            "tree_sitter"
+#                A fast highlighter that produces consistent results as the code
+#                is being edited. However, this highlighter supports only a few
+#                languages. Several things must be done for a language to be
+#                supported, and as a user of Porcupine, you probably shouldn't
+#                try to configure this highlighter yourself; just ask me to do
+#                it instead by making an issue on GitHub.
+#
+#                Recommended for languages with large files or known bugs with
+#                the "pygments" highlighter.
+#
+#        Regardless of this setting, the color theme is always taken from
+#        Pygments, because there are many Pygments themes and it is easy to get
+#        them to work with other highlighting libraries. The theme can be
+#        changed in Porcupine Settings (in the Settings menu).
+#
 #    pygments_lexer (default: "pygments.lexers.TextLexer")
-#        Name of a Pygments lexer class that will be used for syntax
-#        highlighting. See the Pygments documentation for lexers that Pygments
-#        comes with:
+#        If syntax_highlighter is set to "pygments", this should be the name of
+#        a Pygments lexer class that will be used for syntax highlighting. See
+#        the Pygments documentation for lexers that Pygments comes with:
```

Comment:
Old explanation and new explanation are both bad. You should set `pygments_lexer` even if `syntax_highlighting` is set to `"tree_sitter"`, because the pastebin plugin uses the pygments lexer for dpaste.com's syntax highlighting.